openwisp_controller/config/tasks.py

```diff
@@ -97,7 +97,9 @@ def invalidate_devicegroup_cache_delete(instance_id, model_name, **kwargs):
         )
 
 
-@shared_task(base=OpenwispCeleryTask)
+# Generating large configurations can be time-consuming; hence,
+# a custom soft time limit is applied here.
+@shared_task(soft_time_limit=120)
```

Comment:
What about 5 minutes:

```suggestion
@shared_task(soft_time_limit=300)
```

It's reasonable, if it takes longer than this there's most probably something wrong.